domain/secretbackend/params.go

```diff
@@ -4,9 +4,10 @@
 package secretbackend
 
 import (
+	"fmt"
 	"time"
 
-	"github.com/juju/juju/core/secrets"
+	backenderrors "github.com/juju/juju/domain/secretbackend/errors"
 )
 
 // UpsertSecretBackendParams are used to upsert a secret backend.
```

Comment:
ID here is used to lookup the backend to update but that's not how we should be doing it. We need to do the update based on what comes over the wire. The user passes in the name of the backend to update, so as part of the txn we need to look up the uuid of that name. We also need to pass in the "name change" value if the user wants to rename the backend.

Reply:
I think it's wrong to expose the `Upsert_x` from state to service.
The state should be straightforward and uncluttered, avoiding introducing undue complexity or overly "intelligent" behaviors.
State methods should avoid incorporating excessive versatility or implicitly understanding a wide array of arguments. 
Therefore, the state should have separate Create_x and Update_x methods that internally execute the upsert operation. It should be up to the service layer to determine whether to invoke Create_x or Update_x.
By choosing to expose Upsert_x and equipping it to accommodate all the previously mentioned functionalities,
the argument struct might have to be like (I don't like):
```go
type UpsertSecretBackendParams struct {
	ID                  string
	Name                string
	NameChange          *string
	BackendType         string  // We need to validate this for updates because it's an immutable field.
	TokenRotateInterval *time.Duration
	NextRotateTime      *time.Time
	Config              map[string]string
}

```
I would prefer clear and straightforward to make them separately:
```go
type CreateSecretBackendParams struct {
	ID                  string
	Name                string
	BackendType         string
	TokenRotateInterval *time.Duration
	NextRotateTime      *time.Time
	Config              map[string]string
}

type UpdateSecretBackendParams struct {
	ID                  string
	Name                *string
	TokenRotateInterval *time.Duration
	NextRotateTime      *time.Time
	Config              map[string]string
}
```